cmd/cinder_flexvolume_driver

```diff
@@ -0,0 +1,308 @@
+                log "{\"status\": \"Success\"}"
+                exit 0
+        fi
+
+	VOLFSTYPE=`blkid -o udev ${DMDEV} 2>/dev/null|grep "ID_FS_TYPE"|cut -d"=" -f2`
+
+        if [ "${VOLFSTYPE}" == "" ]; then
+                mkfs -t ${FSTYPE} ${DMDEV} >/dev/null 2>&1
+                if [ $? -ne 0 ]; then
+                        err "{ \"status\": \"Failure\", \"message\": \"Failed to create fs ${FSTYPE} on device ${DMDEV}\"}"
+                        exit 1
+                fi
+        fi
+
+        mkdir -p ${MNTPATH} &> /dev/null
+        logfile "Start Mount "  ${DMDEV} ${MNTPATH} 
+
+	mount ${DMDEV} ${MNTPATH} &> /dev/null
+        if [ $? -ne 0 ]; then
+                err "{ \"status\": \"Failure\", \"message\": \"Failed to mount device ${DMDEV} at ${MNTPATH}\"}"
+                exit 1
+        fi
+
+	#
+        # put the flex volume information for virtlet flexvolume resource to consume
+        #
+	echo ${JSON_PARAMS} | jq --arg volpath ${DMDEV} '. + {path: $volpath}' > ${MNTPATH}/virtlet-flexvolume.json
+        # echo ${JSON_PARAMS} > ${MNTPATH}/virtlet-flexvolume.json
+        if [ $? -ne 0 ]; then
+                err "{ \"status\": \"Failure\", \"message\": \"Failed to call mounter""}"
+                exit 1
+        fi
+
+	logfile "Mount completed" 
+        log "{\"status\": \"Success\"}"
+        exit 0
+}
+
+unmountdevice() {
+        logfile "Unmount from local file system" "Args $*"
+        MNTPATH=$1
+
+        if [ ! -d ${MNTPATH} ]; then
+                log "{\"status\": \"Success\"}"
+                exit 0
+        fi
+
+        if [ $(ismounted) -eq 0 ] ; then
+                log "{\"status\": \"Success\"}"
+                exit 0
+        fi
+
+        umount ${MNTPATH} &> /dev/null
+        if [ $? -ne 0 ]; then
+                err "{ \"status\": \"Failed\", \"message\": \"Failed to unmount volume at ${MNTPATH}\"}"
+                exit 1
+        fi
+
+	rm -r -f $MNTPATH &> /dev/null
+        if [ $? -ne 0 ]; then
+                err "{ \"status\": \"Failed\", \"message\": \"Failed to delete directory ${MNTPATH}\"}"
+                exit 1
+        fi
+
+        log "{\"status\": \"Success\"}"
+        exit 0
+}
+
+ismounted() {
+        MOUNT=`findmnt -n ${MNTPATH} 2>/dev/null | cut -d' ' -f1`
+        if [ "${MOUNT}" == "${MNTPATH}" ]; then
+                echo "1"
+        else
+                echo "0"
+        fi
+}
+
+op=$1
+
+if [ "$op" = "init" ]; then
+        logfile "Init start" 
+        log "{\"status\": \"Success\"}"
+        exit 0
+fi
+
+if [ $# -lt 2 ]; then
+        usage
+fi
+
+shift
+
+case "$op" in
+        attach)
+                attach $*
+                ;;
+        detach)
+                detach $*
+                ;;
+        waitforattach)
+                waitforattach $*
+                ;;
+        mountdevice)
+                domountdevice $*
+                ;;
+        unmountdevice)
+                unmountdevice $*
+                ;;
+	unmount)
```

Comment:
What's the difference between "unmount" and "unmountdevice"?  Should "unmount" be removed here? It wasn't listed in the usage text.

Reply:
both of them are flex driver calls. will add usage part.